pages/_app.js

```diff
@@ -1,33 +1,46 @@
-import React from 'react'
-import App from 'next/app'
 import Honeybadger from '@honeybadger-io/js'
 
 // https://docs.honeybadger.io/lib/javascript/reference/configuration.html
-Honeybadger.configure({
+const sharedHoneybadgerConfig = {
   apiKey: process.env.HONEYBADGER_API_KEY,
   revision: process.env.HONEYBADGER_REVISION,
   environment: process.env.NODE_ENV,
   projectRoot: 'webpack://_N_E/./',
-
   // Uncomment to report errors in development:
-  reportData: true,
-})
+  // reportData: true,
+}
+
+if (typeof window === 'undefined') {
+  // Node config
+  const projectRoot = process.cwd()
+  Honeybadger.configure({
+    ...sharedHoneybadgerConfig,
+    projectRoot: 'webpack:///./',
+  }).beforeNotify((notice) => {
+    notice.backtrace.forEach((line) => {
+      if (line.file) {
+        line.file = line.file.replace(`${ projectRoot }/.next/server`, `${process.env.HONEYBADGER_ASSETS_URL}/..`)
```

Comment:
Based on https://github.com/honeybadger-io/nextjs-with-honeybadger/compare/node-source-maps?expand=1#diff-c85e043f1db00b919581c615d87508f7d4ff2bb8058798ebbe475b4a6e087f26, but I had to add `/..` to get sourcemaps working for me. I'm concerned that this might not be universal, however. And this whole thing feels a bit janky... I hope we do not need to ask users to copy/paste this whole thing.

Reply:
This also seems to work when deployed to Heroku, so I am inclined to merge it in. I still think asking users to copy/paste all of this is not ideal, but it's better to have an ugly example that works while we think of something more elegant.

Reply:
> while we think of something more elegant.

I think the solution is to release a nextjs dedicated package that applies this _specific_ configuration.